### PR TITLE
Remove some unreachable code

### PR DESCRIPTION
**Pull request by jcouv**
Relates to https://github.com/dotnet/roslyn/issues/24206

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Symbols/Symbol.cs [hunk 1]
```diff
@@ -1123,53 +1123,6 @@ public ImmutableArray<SymbolDisplayPart> ToDisplayParts(SymbolDisplayFormat form
 
         #region ISymbol Members
 
-        SymbolKind ISymbol.Kind
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Symbol.cs]: >SymbolKind ISymbol.Kind [](start = 8, length = 23)

It looks like the code was reachable through ISymbol.Kind API and it was producing incorrect result at least for Discard and for ErrorType. It feels like we need to add some tests.

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

File: src/Compilers/CSharp/Portable/Symbols/Symbol.cs (hunk 1 above)
Comment (jcouv) [reply src/Compilers/CSharp/Portable/Symbols/Symbol.cs]: Added the tests as discussed.

**Comment by jcouv** [comment]
Thanks
@cston Could you do a second review?

**Pull request merged**